bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/UpdateAccessPolicyCommandTests.cs

```diff
@@ -11,30 +15,290 @@
+                mockServiceAccount.OrganizationId = grantedProject.OrganizationId;
+                policyToReturn = new ServiceAccountProjectAccessPolicy
+                {
+                    Id = data,
+                    GrantedProjectId = grantedProject.Id,
+                    Read = true,
+                    Write = true,
+                    ServiceAccount = mockServiceAccount,
+                    GrantedProject = grantedProject,
+                };
+                break;
+        }
+
+        sutProvider.GetDependency<IAccessPolicyRepository>().GetByIdAsync(data).Returns(policyToReturn);
+
+        await Assert.ThrowsAsync<NotFoundException>(() =>
+            sutProvider.Sut.UpdateAsync(data, read, write, userId));
+        await sutProvider.GetDependency<IAccessPolicyRepository>().DidNotReceiveWithAnyArgs().ReplaceAsync(default);
+    }
+
+    [Theory]
+    [BitAutoData(AccessPolicyType.UserServiceAccountAccessPolicy, PermissionType.RunAsAdmin)]
+    [BitAutoData(AccessPolicyType.UserServiceAccountAccessPolicy, PermissionType.RunAsUserWithPermission)]
+    [BitAutoData(AccessPolicyType.GroupServiceAccountAccessPolicy, PermissionType.RunAsAdmin)]
+    [BitAutoData(AccessPolicyType.GroupServiceAccountAccessPolicy, PermissionType.RunAsUserWithPermission)]
+    public async Task UpdateAsync_ServiceAccountGrants_PermissionsCheck_Success(
+        AccessPolicyType accessPolicyType,
+        PermissionType permissionType,
+        Guid data,
+        bool read,
+        bool write,
+        Guid userId,
+        ServiceAccount grantedServiceAccount,
+        Group mockGroup,
+        SutProvider<UpdateAccessPolicyCommand> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().AccessSecretsManager(Arg.Any<Guid>()).Returns(true);
+        BaseAccessPolicy policyToReturn = null;
+        switch (accessPolicyType)
+        {
+            case AccessPolicyType.UserServiceAccountAccessPolicy:
+                policyToReturn =
+                    new UserServiceAccountAccessPolicy
+                    {
+                        Id = data,
+                        Read = true,
+                        Write = true,
+                        GrantedServiceAccountId = grantedServiceAccount.Id,
+                        GrantedServiceAccount = grantedServiceAccount,
+                    };
+                break;
+            case AccessPolicyType.GroupServiceAccountAccessPolicy:
+                mockGroup.OrganizationId = grantedServiceAccount.OrganizationId;
+                policyToReturn =
+                    new GroupServiceAccountAccessPolicy
+                    {
+                        Id = data,
+                        GrantedServiceAccountId = grantedServiceAccount.Id,
+                        Read = true,
+                        Write = true,
+                        Group = mockGroup,
+                        GrantedServiceAccount = grantedServiceAccount,
+                    };
+                break;
+        }
```

Comment:
Extract to function?

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/79bce819b50d75b5ce021b4b4dfad21dc9c97cd8